src/integration-test/java/com/commercetools/sync/integration/commons/utils/CategoryITUtils.java

```diff
@@ -0,0 +1,345 @@
+     * children and grand children.
+     *
+     *
+     * @param ctpClient defines the CTP project to delete the categories from.
+     */
+    public static void deleteAllCategories(@Nonnull final SphereClient ctpClient) {
+        final Set<String> keys = new HashSet<>();
+        final List<Category> categories = QueryExecutionUtils.queryAll(ctpClient,
+            CategoryQuery.of().withExpansionPaths(CategoryExpansionModel::ancestors))
+                                                             .thenApply(CategoryITUtils::sortCategoriesByLeastAncestors)
+                                                             .toCompletableFuture().join();
+        categories.forEach(category -> {
+            final String categoryKey = category.getKey();
+            if (!hasADeletedAncestor(category, keys)) {
+                ctpClient.execute(CategoryDeleteCommand.of(category))
+                         .thenAccept(deletedCategory -> keys.add(categoryKey))
+                         .toCompletableFuture().join();
+            }
+        });
+    }
+
+    private static List<Category> sortCategoriesByLeastAncestors(@Nonnull final List<Category> categories) {
+        categories.sort(Comparator.comparingInt(category -> category.getAncestors().size()));
+        return categories;
+    }
+
+    private static boolean hasADeletedAncestor(@Nonnull final Category category,
+                                               @Nonnull final Set<String> keysOfDeletedAncestors) {
+        final List<Reference<Category>> categoryAncestors = category.getAncestors();
+        return categoryAncestors.stream().anyMatch(ancestor ->
+            keysOfDeletedAncestors.contains(ancestor.getObj().getKey()));
+    }
+
+
+    /**
+     * Given a list of {@link CategoryDraft} batches represented by a
+     * {@link List}&lt;{@link List}&lt;{@link CategoryDraft}&gt;&gt; and an instance of {@link CategorySync}, this
+     * method recursively calls sync by the instance of {@link CategorySync} on each batch, then removes it, until
+     * there are no more batches, in other words, all batches have been synced.
+     *
+     * @param categorySync the categorySync instance to sync with each batch of {@link CategoryDraft}
+     * @param batches      the batches of {@link CategoryDraft} to sync.
+     * @param result       in the first call of this recursive method, this result is normally a completed future, it
+     *                     used from within the method to recursively sync each batch once the previous batch has
+     *                     finished syncing.
+     * @return an instance of {@link CompletionStage} which contains as a result an instance of
+     *          {@link CategorySyncStatistics} representing the {@code statistics} of the sync process executed on the
+     *          given list of batches.
+     */
+    public static CompletionStage<CategorySyncStatistics> syncBatches(@Nonnull final CategorySync categorySync,
```

Comment:
I see that result param is not nullable. It means that the whole method can return a null. Isn't it better to note the method as a nullable?